### PR TITLE
fix: Handles button binding during buildingblock pasting

### LINKED ISSUE #33658: Bug: Binding issue with insert data building block
When we drop multiple Insert data building blocks, UI bindings(i.e. from one widget to another) are not getting updated correctly.

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
This pull request adds the handleButtonDynamicTriggerPathList function to the BuildingBlockAdditionSagas.ts file. This function is responsible for handling the dynamic trigger path list for button widgets. It iterates over the widgetNameMap and updates the dynamicTriggerPathList of each button widget with the newWidgetName. 

Additionally, unit tests have been added to ensure the correct functionality of the handleButtonDynamicTriggerPathList function. This change improves the functionality of button widgets when pasting building block widgets.

Fixes #33658 
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Widget"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9268894716>
> Commit: a675a5c8a752fea76c130e8bdc1b76e44453ee98
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9268894716&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by jacquesikot — APPROVED** [review]

File: app/client/src/widgets/wds/WDSCurrencyInputWidget/config/autocompleteConfig.ts [hunk 1]
```diff
@@ -4,12 +4,12 @@ export const autocompleteConfig = {
   "!doc":
     "An input text field is used to capture a currency value. Inputs are used in forms and can have custom validations.",
   "!url": "https://docs.appsmith.com/widget-reference/currency-input",
-  text: {
+  parsedText: {
```
Comment (marks0351) [inline app/client/src/widgets/wds/WDSCurrencyInputWidget/config/autocompleteConfig.ts:7]: @KelvinOm @jsartisan can you guys review these changes to wds widgets.

File: app/client/src/pages/Editor/IDE/Sidebar/SidebarComponent.tsx [hunk 2]
```diff
@@ -23,6 +26,23 @@ interface SidebarComponentProps {
...
+        if (datasources.length === 0)
+          return {
+            conditionIcon: "warning",
+            tooltip: conditionTooltip,
+          };
+        return {};
+      default:
+        return {};
+    }
+  };
```
[the remark this answers is not in the document]
Comment (marks0351) [reply app/client/src/pages/Editor/IDE/Sidebar/SidebarComponent.tsx:45]: This comment does make sense.

File: app/client/src/widgets/wds/WDSCurrencyInputWidget/config/autocompleteConfig.ts (hunk 1 above)
Comment (rahulbarwal) [reply app/client/src/widgets/wds/WDSCurrencyInputWidget/config/autocompleteConfig.ts:7]: Sorry folks this is due to some random rebase of release on my PR. 
These are not my changes. No need to review this PR, I have for now put it in draft state, but may remove it.

So sorry @marks0351

**Review by marks0351 — COMMENTED** [review]
I think some of the changes are of other branches. could you pls merge latest release

**Pull request merged**